### PR TITLE
[JIT] Lazily initialize aliasDb in DCE

### PR DESCRIPTION
**Pull request by desertfire**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#56649 [JIT] Lazily initialize aliasDb in DCE**

Differential Revision: [D27926043](https://our.internmc.facebook.com/intern/diff/D27926043)

### CODE REVIEW

File: torch/csrc/jit/passes/dead_code_elimination.cpp [hunk 1]
```diff
@@ -218,11 +219,12 @@ class DeadCodeEliminator {
       }
     }
 
-    if (aliasDb_) {
-      if (aliasDb_->writesToAlias(node, liveValues_)) {
+    if (useAliasDb_) {
```
Comment (penguinwu) [inline torch/csrc/jit/passes/dead_code_elimination.cpp:222]: In the old code, `aliasDb_` is always initialized, so I don't understand why the old code needs to check `if(aliasDb_)`, perhaps an unnecessary checking? If so, then we can remove `useAliasDb_` altogether.

**Review by penguinwu — CHANGES_REQUESTED** [review]
LGTM, just one comment on the checking of `if aliasDb_` from the old code. We may be able to remove that redundant code, perhaps check with Elias.

File: torch/csrc/jit/passes/dead_code_elimination.cpp (hunk 1 above)
Comment (desertfire) [reply torch/csrc/jit/passes/dead_code_elimination.cpp:222]: There are two constructors for DeadCodeEliminator, one initializes aliasDb_ and the other doesn't. Not sure why, but that's why I decided to use useAliasDb_ to capture that.

**Review by penguinwu — APPROVED** [review]
LGTM

**Comment by desertfire** [comment]
@desertfire has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D27926043).

**Pull request closed without merging**